### PR TITLE
Ci lint setup

### LINKED ISSUE #26: Add GitHub apps and integrations
- [x] Something to run our linting - consider actual *writing* lint results
- [x] Code coverage integrating with istanbul
- [x] CircleCI
- [x] PR review system
- [x] Autobuild documentation site. 
- [x] Remove snyk because GH vulnerability alerts

### PR DESCRIPTION
**Pull request by bencooper222**
This PR does a few big things. 
- Changes how our eslint commands work by taking advantage of how eslint allow folder-scoped config files to override project scoped config files. 
- Changes our eslint config to eliminate duplicated and unnecessary config. @leonm1 please be aggressive in making sure that I replicated the functionality of the old config
- Redoes our CircleCI config to include linting. It currently is *not* required to merge a PR so I'm comfortable introducing it. 

~~Please hold on reviewing this until I assign - I want to see how CircleCI handles the new config files.~~

Fixes: #42
Fixes: #77
Fixes #26

### CODE REVIEW

**Comment by cktang88** [comment]
lol what are these commit messages

**Comment by bencooper222** [comment]
*Finally* fucking done.
Related: https://twitter.com/bcoops222/status/1121160370037477377

File: src/client/.eslintrc.js [hunk 1]
```diff
@@ -37,7 +37,6 @@ module.exports = {
 				extensions: ['.tsx', '.jsx'],
 			},
 		],
-		'sort-keys-fix/sort-keys-fix': ['error', 'asc', { caseSensitive: true, natural: true }],
```
Comment (cktang88) [inline src/client/.eslintrc.js]: why remove sort-keys?

Comment (bencooper222) [reply src/client/.eslintrc.js]: Moves to top level eslint config because it's a common rule.

**Comment by leonm1** [comment]
I'm not sure if we completely agreed on git commit messages, but here's the gold standard: https://chris.beams.io/posts/git-commit/ I think it would behoove us to `git rebase -i` and rewrite some of these commit messages to make it more clear than `i hate yaml` and `yaml is a scam`

**Comment by bencooper222** [comment]
So I switched to ssh remote in the middle of this and I'm pretty sure the history is truly fucked right now 😢

**Comment by bencooper222** [comment]
JK I think I unfucked it?

**Comment by bencooper222** [comment]
Let's fix #71 before we merge this

File: src/server/.eslintrc.js [hunk 2]
```diff
@@ -1,32 +1,15 @@
...
+			'@typescript-eslint/parser': ['.ts', '.tsx'],
+		},
 		'import/resolver': {
-			webpack: {
-				config: 'webpack.dev.js',
-			},
+			typescript: {},
 		},
 	},
-	plugins: ['sort-keys-fix'],
```
Comment (cktang88) [inline src/server/.eslintrc.js]: why are we removing this?

File: src/server/index.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ const app = new koa();
 const router = new koaRouter();
 
 // Default port to listen
-const port = 8080;
```
Comment (cktang88) [inline src/server/index.ts]: This file contains changes already in `dev` branch, not related to PR

**Comment by bencooper222** [comment]
The more I think about this, this is quickly becoming too monolithic of a PR (that's on me). I think we should merge in #97 to this, then merge this to `dev` and let PRs like #89 be merged straight into dev. Sound like a plan @tsaok98 @cktang88 @leonm1 ?

**Pull request merged**